pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/column/PhysicalColumnIndexContainer.java

```diff
@@ -95,10 +95,24 @@ public PhysicalColumnIndexContainer(SegmentDirectory.Reader segmentReader, Colum
       _nullValueVectorReader = null;
     }
 
-    if (loadTextIndex && segmentIndexDir != null) {
-      Preconditions.checkState(segmentReader.hasIndexFor(columnName, ColumnIndexType.TEXT_INDEX));
+    if (loadTextIndex
+        && segmentIndexDir != null) {
+      boolean hasIndexFor = segmentReader.hasIndexFor(columnName, ColumnIndexType.TEXT_INDEX);
       Map<String, Map<String, String>> columnProperties = indexLoadingConfig.getColumnProperties();
-      _textIndex = indexReaderProvider.newTextIndexReader(segmentIndexDir, metadata, columnProperties.get(columnName));
+      if (processExistingSegments(columnName, columnProperties)) {
+        Preconditions.checkState(hasIndexFor);
+        _textIndex = indexReaderProvider.newTextIndexReader(segmentIndexDir, metadata,
+            columnProperties.get(columnName));
+      } else if (hasIndexFor) {
```

Comment:
When will this happen for existing segments - `processExistingSegments` returned false but `hasIndexFor` is true?

Reply:
The logic here is:
- If the column is not configured with property - `skipExistingSegments`,  existing code flow.
- If the column is configured with `skipExistingSegments`, we check if `lucene` index already exists, then read it for the segments. And, if it doesn't exists, we discard reading it.